### DISCUSSION TITLE
Hook file as layout implementation

### DISCUSSION THREAD

**Discussion by luist1228** [sveltejs/kit#5402] [Ideas]
I don't know if this fits withing the scope of the project but having hooks as a top thing that happens before calling your endpoint feels a bit limiting. The idea here is to have a `__hook.ts` file wich has a `handle()` just like the normal hooks file, I think this way would be more flexible and granular how we control our endpoints.

``` 
routes/
├─ __hook.ts
├─ __layout.svelte
├─ api/
│  ├─ private/
│  │  ├─ index.ts
│  │  ├─ __hook.ts
│  │  ├─ sub_private/
│  │  │  ├─ index.ts
│  ├─ public/
│  │  ├─ index.ts
```    

When we call `fetch('api/private/sub_private')` every `__hook.ts` that's before the endpoint in the routes tree would be called in order and for example if there is some auth or role based authorization verfication it can be done closer to the endpoint call.

I don't know if this is too much like an express middleware model or if this could fit well within sveltekit, but it's an idea that I had on my mind for a while.

**Comment by benmccann** [comment]
This sounds a bit like https://github.com/sveltejs/kit/issues/4274

**Comment by TranscendentalRide** [comment]
I'm a newbie here--having one top-level hook endpoint is a bit limiting than having route-based hooks. One obvious application is authentication--some parts of the app need authentication, or have role restrictions, and other parts are public. 

Maybe a `+hook.ts` or something like that...
